File: js/functions/helpers/filterFindWords.js
function filterFindWords(searchWordList) {
    const findWordsList = document.querySelectorAll('.find-words-list__item');
    let notFoundWordsList = null;

    if (findWordsList.length > 0) {
        notFoundWordsList = searchWordList.filter(word => {
            return !Array.from(findWordsList).some(findWord => word === findWord.innerHTML.toLowerCase());
        });
        notFoundWordsList = notFoundWordsList[0];
    } else {
        notFoundWordsList = searchWordList[0];
    }

    return notFoundWordsList;
}

export default filterFindWords;

File: cpp/filterFindWords.cpp
#include <iostream>
#include <string>
#include <vector>

std::string filterFindWords(const std::vector<std::string>& searchWordList, const std::vector<std::string>& findWordsList) {
    std::string notFoundWordsList;

    if (findWordsList.size() > 0) {
        std::vector<std::string> tempNotFoundWordsList;
        for (const auto& word : searchWordList) {
            bool found = false;
            for (const auto& findWord : findWordsList) {
                if (word == findWord) {
                    found = true;
                    break;
                }
            }
            if (!found) {
                tempNotFoundWordsList.push_back(word);
            }
        }

        if (!tempNotFoundWordsList.empty()) {
            notFoundWordsList = tempNotFoundWordsList[0];
        }
    } else {
        notFoundWordsList = searchWordList[0];
    }

    return notFoundWordsList;
}